### PR TITLE
DPP 0.21 Updates

### PR DESCRIPTION
**Pull request by HashEngineering**

### CODE REVIEW

File: src/main/kotlin/org/dashj/platform/dpp/errors/concensus/ConcensusException.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                Codes.DocumentAlreadyPresentError -> DocumentAlreadyPresentException(arguments)
+//                Codes.DocumentNotFoundError -> DocumentNotFoundException(arguments)
+//                Codes.DocumentOwnerIdMismatchError -> DocumentOwnerIdMismatchException(arguments)
+//                Codes.DocumentTimestampsMismatchError -> DocumentTimestampsMismatchException(arguments)
+//                Codes.DocumentTimestampWindowViolationError -> DocumentTimestampWindowViolationException(arguments)
+//                Codes.DuplicateUniqueIndexError -> DuplicateUniqueIndexException(arguments)
+                Codes.InvalidDocumentRevisionError -> InvalidDocumentRevisionException(arguments)
+                Codes.IdentityAlreadyExistsError -> IdentityAlreadyExistsException(arguments)
+//                Codes.IdentityPublicKeyAlreadyExistsError -> IdentityPublicKeyAlreadyExistsException(arguments)
+                else -> throw UnknownConcensusError(code.code, arguments)
```
Comment (Syn-McJ) [inline src/main/kotlin/org/dashj/platform/dpp/errors/concensus/ConcensusException.kt:124]: All those commented errors, are they gonna be used in the future?

File: src/test/kotlin/org/dashj/platform/dpp/Fixtures.kt [hunk 2]
```diff
@@ -1,86 +1,332 @@
...
+        data.putAll(options)
+        return documentFactory.create(dataContract, ownerId, "preorder", data)
+    }
+
+    fun getInstantAssetLockProofFixture(oneTimePrivateKey: ECKey = ECKey()): InstantAssetLockProof {
+        val privateKeyHex = "cSBnVM4xvxarwGQuAfQFwqDg9k5tErHUHzgWsEfD4zdwUasvqRVY"
+        val privateKey = ECKey.fromPrivate(Base58.decode(privateKeyHex))
+        // val fromAddress = Address.fromKey(PARAMS, privateKey)
+
+        // val oneTimePublicKey = oneTimePrivateKey.pubKey
```
Comment (Syn-McJ) [inline src/test/kotlin/org/dashj/platform/dpp/Fixtures.kt:271]: Admittedly, I don't entirely understand the purpose of this code (is this for tests only?), but should there be private keys hardcoded? Also, commented code - better to either clean up or leave some explanation.

**Review by Syn-McJ — APPROVED** [review]
Looks good to me, there are a couple of questions I have that are probably not critical.

File: src/main/kotlin/org/dashj/platform/dpp/errors/concensus/ConcensusException.kt (hunk 1 above)
Comment (HashEngineering) [reply src/main/kotlin/org/dashj/platform/dpp/errors/concensus/ConcensusException.kt:124]: Some of them will.  Currently, the ones that are not used by any other layer of our system are commented out.

File: src/test/kotlin/org/dashj/platform/dpp/Fixtures.kt (hunk 2 above)
Comment (HashEngineering) [reply src/test/kotlin/org/dashj/platform/dpp/Fixtures.kt:271]: Yes, the `Fixtures.kt` file is for tests only.  Some private keys will be hard coded in the tests to sign data and to verify signatures.  These tests are taken from the `js-dpp` project, which is the Dash Platform Protocol repo maintained by the Platform team.

I will search through the current code and clean out commented code.

**Pull request merged**